tsec-libsodium/src/main/scala/tsec/libsodium/authentication/OnetimeAuth.scala

```diff
@@ -0,0 +1,48 @@
+package tsec.libsodium.authentication
+
+import cats.effect.Sync
+import tsec.libsodium.ScalaSodium
+import tsec.libsodium.cipher.SodiumKey
+
+sealed trait OnetimeAuth
+
+sealed trait MessageTag
+
+object OnetimeAuth {
+
+  case object InvalidMessage extends Exception with Product with Serializable {
```

Comment:
`with Product with Serializable` is unneeded I think

Reply:
I think this style is used over the complete code base. Should be change this everywhere ?

Reply:
Maybe. @jmcardon was there a reason?